docs/operator-manual/applicationset/Generators-Plugin.md

```diff
@@ -0,0 +1,237 @@
+# Plugin Generator
+
+Plugins allow you to provide your own generator.
+
+* You can write in any language
+* Simple: a plugin just responds to RPC HTTP requests.
+* You can use it in a sidecar, or standalone deployment.
+* You can get your plugin running today, no need to wait 3-5 months for review, approval, merge and an Argo software
+  release.
+* You can combine it with Matrix or Merge.
+
+## Simple example
+
+Using a generator plugin without combining it with Matrix or Merge.
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: ApplicationSet
+metadata:
+  name: myplugin
+spec:
+  generators:
+    - plugin:
+        # The name of the plugin used
+        name: my-plugin
+        # Specify the configMap where the plugin configuration is located
+        configMapRef: my-plugin
+        # You can pass parameters included in the RPC call
+        params:
+          key1: "value1"
+          key1: "value1"
+        # When using a Plugin generator, the ApplicationSet controller polls every `requeueAfterSeconds` interval (defaulting to every 30 minutes) to detect changes.
+        requeueAfterSeconds: 30
+        # ...
+```
+
+* `name`: Required name of the Plugin.
+* `configMapRef`: A `ConfigMap` name containing the Plugin configuration to use for RPC call.
+* `params`: Parameters included in the RPC call. (Optional)
+
+!!! note
+    The concept of the plugin should not undermine the spirit of GitOps by externalizing data outside of Git. The goal is to be complementary in specific contexts.
+    For example, when using one of the PullRequest generators, it's impossible to retrieve parameters related to the CI (only the commit hash is available), which limits the possibilities. By using a plugin, it's possible to retrieve the necessary parameters from a separate data source and use them to extend the functionality of the generator.
+
+### Add a ConfigMap to configure the access of the plugin
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-plugin
+data:
+  token: $plugin.myplugin.token # Alternatively $<some_K8S_secret>:plugin.myplugin.token
```

Comment:
If the plugin Deployment lives outside the `argocd` namespace, I think we'll have to duplicate the token. It's not possible to mount a secret from the `argocd` namespace (and probably is not desirable). I think this is a reasonable UX, we might just want to document it.